include/seqan3/range/view/async_buffer.hpp

```diff
@@ -0,0 +1,538 @@
+            {
+                switch (contrib::queue_op_status s = buffer_ptr->try_pop(cached_value); s)
+                {
+                    case contrib::queue_op_status::closed:
+                        at_end = true;
+                        return *this;
+                    case contrib::queue_op_status::success:
+                        /* cached_value was updated */
+                    #ifndef NDEBUG
+                        has_value = true;
+                    #endif
+                        return *this;
+                    default:
+                        assert(s == contrib::queue_op_status::empty);
+                        break; // loop again
+                }
+
+                delay.wait(); // pause and then try again.
+            }
+
+            // TODO [[unreachable]]
+            return *this;
+        }
+
+        //!\brief Post-increment.
+        void operator++(int) noexcept
+        {
+            ++(*this);
+        }
+        //!\}
+
+        /*!\name Comparison operators
+        * \{
+        */
+        //!\brief Compares for equality with sentinel.
+        friend constexpr bool operator==(async_buffer_iterator const & lhs,
+                                         std::ranges::default_sentinel_t const &) noexcept
+        {
+            return lhs.at_end;
+        }
+
+        //!\copydoc operator==
+        friend constexpr bool operator==(std::ranges::default_sentinel_t const &,
+                                         async_buffer_iterator const & rhs) noexcept
+        {
+            return rhs == std::ranges::default_sentinel_t{};
+        }
+
+        //!\brief Compares for inequality with sentinel.
+        friend constexpr bool operator!=(async_buffer_iterator const & lhs,
+                                         std::ranges::default_sentinel_t const &) noexcept
+        {
+            return !(lhs == std::ranges::default_sentinel_t{});
+        }
+
+        //!\copydoc operator!=
+        friend constexpr bool operator!=(std::ranges::default_sentinel_t const &,
+                                         async_buffer_iterator const & rhs) noexcept
+        {
+            return rhs != std::ranges::default_sentinel_t{};
+        }
+        //!\}
+    };
+
+public:
+    /*!\name Constructor, destructor, and assignment.
+     * \{
+     */
+    async_buffer_view()                                         = default; //!< Defaulted.
+    async_buffer_view(async_buffer_view const &)                = delete;  //!< Deleted.
+    async_buffer_view(async_buffer_view &&)                     = default; //!< Defaulted.
+    async_buffer_view & operator=(async_buffer_view const &)    = delete;  //!< Deleted.
+    async_buffer_view & operator=(async_buffer_view &&)         = default; //!< Defaulted.
+
+    /*!\brief Tell thread to stop and wait for it.
+     *
+     * \details
+     *
+     * Destruction of the view clears the buffer and temporary storage. Up to `n + 1` elements may be deleted
```

Comment:
Is deleted the correct word? 
When I have a ForwardRange and I read *deleted* I would kinda expect that it is really deleted - e.g. if I have a container, the size changes. But actually it's just *moved*, so the positions where the elements where are now still accessible and contain valid, but undefined stuff, right?

Reply:
Well, yes,  "moved-from objects shall be placed in a valid but unspecified state."

The elements are destructed, though. I can write that instead of deleted.